exporters/trace/zipkin/zipkin.go

```diff
@@ -35,58 +40,116 @@ type Exporter struct {
 	serviceName string
 	client      *http.Client
 	logger      *log.Logger
+	o           options
 }
 
 var (
 	_ export.SpanBatcher = &Exporter{}
 )
 
 // Options contains configuration for the exporter.
-type Options struct {
-	client *http.Client
-	logger *log.Logger
+type options struct {
+	serviceName string
+	client      *http.Client
+	logger      *log.Logger
+	config      *sdktrace.Config
 }
 
 // Option defines a function that configures the exporter.
-type Option func(*Options)
+type Option func(*options)
+
+// WithServiceName sets the exporter to use the passed service name.
+func WithServiceName(serviceName string) Option {
+	return func(opts *options) {
+		opts.serviceName = serviceName
+	}
+}
 
 // WithLogger configures the exporter to use the passed logger.
 func WithLogger(logger *log.Logger) Option {
-	return func(opts *Options) {
+	return func(opts *options) {
 		opts.logger = logger
 	}
 }
 
 // WithClient configures the exporter to use the passed HTTP client.
 func WithClient(client *http.Client) Option {
-	return func(opts *Options) {
+	return func(opts *options) {
 		opts.client = client
 	}
 }
 
-// NewExporter creates a new zipkin exporter.
-func NewExporter(collectorURL string, serviceName string, os ...Option) (*Exporter, error) {
-	if _, err := url.Parse(collectorURL); err != nil {
+// WithSDK sets the SDK config for the exporter pipeline.
+func WithSDK(config *sdktrace.Config) Option {
+	return func(o *options) {
+		o.config = config
+	}
+}
+
+// NewRawExporter creates a new Zipkin exporter.
+func NewRawExporter(collectorURL string, opts ...Option) (*Exporter, error) {
+	if collectorURL == "" {
+		return nil, errors.New("collector URL cannot be empty")
+	}
+	u, err := url.Parse(collectorURL)
+	if err != nil {
 		return nil, fmt.Errorf("invalid collector URL: %v", err)
 	}
-	if serviceName == "" {
-		return nil, fmt.Errorf("service name must be non-empty string")
+	if u.Scheme == "" || u.Host == "" {
+		return nil, errors.New("invalid collector URL")
+	}
+
+	o := options{}
+	for _, opt := range opts {
+		opt(&o)
 	}
-	opts := Options{}
-	for _, o := range os {
-		o(&opts)
+	if o.serviceName == "" {
+		o.serviceName = defaultServiceName
```

Comment:
Having a default service name doesn't seem correct. The zipkin model need a unique service name to delineate services in its representation of the distributed trace. With a design that defaults to a common service name it seems like it will lead to a class of error and frustration for the user where a "successful" setup and run of the system will contain invalid data.

It might make sense to have an exception here to the similar function signatures across the exporters. Given Zipkin has different requirements, having the service name be an argument would make sense to me.

Reply:
Understood, in that case this was an oversight on my part. Keeping it as an argument makes perfect sense then.